### ISSUE TITLE
Docs build wrongly reports errors

### ISSUE THREAD

**Issue by josephjclark** [OpenFn/adaptors#684]
If you have a typedef before a signature:
```
/**
 * Options provided to the Asana API request
 * @typedef {Object} RequestOptions
 * ...
 */

/**
 * Make a HTTP request against the Asana API.
 * @public
 * ...
 */
export function request(path, params = {}, callback) {
```
The docs build will report an error:
```
Warning: Multiple leading comment blocks found. Discarding all but first:
...
Warning: Invalid documentation for request ✗ Stated params are '', but detected params are '["path","params","callback"]'.
```

Basically it ignores the ACTUAL type def.

This is totally bogus, the generated docs are fine.

A quick fix might be to use the LAST comment block, not the first?

**Comment by josephjclark** [comment]
@taylordowns2000 has rightly flagged that this is a way bigger problem than I realised!

If you do `@state` docs and then a function with JS docs, the builder will associate the state docs with the function, not the actual docs. Because it _discards_ the first. It doesn't use the first.

I suspect I haven't noticed this because `execute` tends to be the first function.

The workaround is to declare typedefs at the bottom of hte file

The fix is to:
a) absorb simple-ast (because it's unmaintainable right now)
b) refine the comment association. We should ignore `@state` and `@typedef` blocks

**Comment by taylordowns2000** [comment]
yeah simple-ast should definitely live in `kit` and i'm happy to answer questions about how/why it does what it does. @stuartc made a fairly clean implementation in March 2021 and I've added 6000+ lines since 😂

**Comment by josephjclark** [comment]
docs.openfn.org is absolutely fine with this. the jsdocs build correctly, as does the adaptor docs panel in lightning. It's just simple-ast that's affected. And so far as I know there isn't anything in v2 that uses simple ast?
